src/SonarScanner.MSBuild.Tasks/IsTestFileByName.cs

```diff
@@ -68,52 +66,29 @@ public sealed class IsTestFileByName : Task
 
         public override bool Execute()
         {
-            var taskSuccess = true;
-
-            var config = TaskUtilities.TryGetConfig(AnalysisConfigDir, new MSBuildLoggerAdapter(Log));
-
-            if (config != null)
+            if (TaskUtilities.TryGetConfig(AnalysisConfigDir, new MSBuildLoggerAdapter(Log)) is AnalysisConfig config)
             {
-                var regEx = TryGetRegularExpression(config);
-
-                try
+                if (config.GetAnalysisSettings(true).TryGetValue(TestRegExSettingId, out var regEx) && !string.IsNullOrWhiteSpace(regEx))
                 {
-                    // Let's use a case sensitive regex (default behavior)
-                    IsTest = Regex.IsMatch(FullFilePath, regEx);
+                    try
+                    {
+                        // Let's use a case sensitive regex (default behavior)
+                        IsTest = Regex.IsMatch(FullFilePath, regEx);
+                    }
+                    catch (ArgumentException ex) // thrown for invalid regular expressions
+                    {
+                        Log.LogError(Resources.IsTest_InvalidRegularExpression, regEx, ex.Message, TestRegExSettingId);
```

Comment:
Presumably this is still sufficient to fail the task if the regex is invalid?

Reply:
previously it was returning `!Log.HasLoggedErrors & taskSuccess`. 

`taskSuccess` seems redundant to me, too.

Reply:
Yes, this should still fail the task. I removed redundant state. There's original `IsTestFile_InvalidRegexInConfig` test for it.